File: WireGuardClientLib/Sources/WireGuardClientLib/WireGuardInterface.swift
import Foundation

public protocol WireGuardInterface {
    func connect(wgQuickConfig: String, completion: @escaping (Result<Void, Error>) -> Void)
    func disconnect(completion: @escaping (Result<Void, Error>) -> Void)
    func connectionStatus() -> WireGuardConnectionStatus
    func trafficStats() -> WireGuardStats
    func pingServer(completion: @escaping (Result<TimeInterval, Error>) -> Void)
}

File: WireGuardClientLibAndroid/app/src/main/java/com/example/wireguardclientlib/WireGuardInterface.java
package com.example.wireguardclientlib;

public interface WireGuardInterface {
    void connect(WireGuardConfiguration configuration, ConnectionCallback callback);
    void disconnect(ConnectionCallback callback);
    WireGuardConnectionStatus getStatus();
    WireGuardStats getStats();

    interface ConnectionCallback {
        void onSuccess();
        void onFailure(WireGuardError error);
    }
}
void pingServer(String host, PingCallback callback);

interface PingCallback {
    void onSuccess(double latencyMs);
    void onFailure(WireGuardError error);
}
